Source/URLDispatcher.spoon/init.lua

```diff
@@ -0,0 +1,83 @@
+--- === URLDispatcher ===
+---
+--- Flexible URL handling
+---
+--- Download: [https://github.com/Hammerspoon/Spoons/raw/master/Spoons/URLDispatcher.spoon.zip](https://github.com/Hammerspoon/Spoons/raw/master/Spoons/URLDispatcher.spoon.zip)
+---
+--- Sets Hammerspoon as the default browser for HTTP/HTTPS links, and
+--- dispatches them to different apps according to the patterns defined
+--- in the config. If no pattern matches, `default_handler` is used.
+
+local obj={}
+obj.__index = obj
+
+-- Metadata
+obj.name = "URLDispatcher"
+obj.version = "0.1"
+obj.author = "Diego Zamboni <diego@zzamboni.org>"
+obj.homepage = "https://github.com/Hammerspoon/Spoons"
+obj.license = "MIT - https://opensource.org/licenses/MIT"
+
+--- URLDispatcher.default_handler
+--- Variable
+--- Bundle ID for default URL handler
+obj.default_handler = "com.apple.Safari"
+
+--- URLDispatcher.decode_slack_redir_urls
+--- Variable
+--- If true, handle Slack-redir URLs to apply the rule on the destination URL
+obj.decode_slack_redir_urls = true
+
+--- URLDispatcher.url_patterns
+--- Variable
+--- URL dispatch rules.
+--- Evaluated in the order they are declared. Entry format: { "url pattern", "application bundle ID" }
```

Comment:
We tend to try to be explicit about the datatypes involved, I would say something like:

```
A table containing dispatch rules. Each rule should be its own table in the format: ...
```

and also this ought to mention that the patterns are Lua patterns, otherwise people might guess they're regexps.